plugins/custom/docker_custom.py

```diff
@@ -181,13 +181,23 @@ def _read_task_stats(self):
                                  humanize_bytes(net_r),
                                  humanize_bytes(net_w)))
 
-            if cpu_percent < 5:
-                unhealthy_count += 1
-            else:
-                unhealthy_count = 0
+            try:
+                 info = self.cli.inspect_container(container=self.container['Id'])
+                 health_check_ok = info['State']['Health']['Status']
+                 if health_check_ok:
+                    unhealthy_count = 0
+                 else:
+                    unhealthy_count += 1
+                    self.log.info('Health check Failed')
+            except:
```

Comment:
This is minor, but do you know what kind of exception it causes if the container doesn't have a HEALTHCHECK command? I could imagine something like the dict lookup calls raising an exception (e.g., if the api keys change) and the log message wouldn't say anything about it. It would be a graceful failure since it would just behave like the old qos check.

Reply:
If the api stays the same, `inspect_container` should always return a dict, it might raise a `docker.errors.APIError` error if the docker server fails. If the container does not have a health check, the dict look up will fail with the standard `KeyError` I suppose. I decided to use the pokemon catch to make sure it always fallback to the old approach on any error, and the user sometimes needs to check the worker log to figure out which one is used.

Reply:
Ok. There might be a small danger if a user assumes that their containers die because of the health check when it's really because they were idle, but I'm ok with this behavior if you are.

Reply:
I added another commit to check if the worker image has builtin health check, and print a message to slack. Can you review that one as well.
It is harder to notify the user the cause when the workers are killed, in order to do that we need to propagate the error from the docker operator back to the manager. Not sure if there is a clean way to do it.
I could not find a python operator in the synaptor dag to insert the `health_check_info` call, so I only modified segmentation dag and chunkflow dag.